src/main/kotlin/com/cognifide/gradle/aem/pkg/deploy/HtmlResponse.kt

```diff
@@ -49,5 +52,48 @@ abstract class HtmlResponse(private val rawHtml: String) {
 
     val success: Boolean
         get() = status == Status.SUCCESS
+
+    companion object {
+        private const val NUMBER_OF_LINES_TO_READ = 5000
+
+        private const val ERROR_SEPARATOR = "\n\n"
+
+        private const val LINE_FEED = 10.toChar()
+
+        fun readFrom(stream: InputStream, errorPatterns: List<ErrorPattern>, statusTags: List<String>): String {
+            val buf = IOUtils.toBufferedInputStream(stream)
+            val reader = buf.bufferedReader()
+            return readByLines(reader,errorPatterns,statusTags)
+        }
+
+        private fun readByLines(input: BufferedReader, errorPatterns: List<ErrorPattern>, statusTags: List<String>): String {
+            val resultBuilder = StringBuilder()
+            val chunk = StringBuilder()
+            var currentLine = 0
+            input.forEachLine {
+                chunk.append(it + LINE_FEED)
+                currentLine++
+                if (currentLine % NUMBER_OF_LINES_TO_READ == 0) {
+                    extractErrors(chunk, resultBuilder, errorPatterns, statusTags)
+                }
+            }
+            extractErrors(chunk, resultBuilder, errorPatterns, statusTags)
+            return resultBuilder.toString()
+        }
+
+        private fun extractErrors(chunk: StringBuilder, builder: StringBuilder, errorPatterns: List<ErrorPattern>, statusTags: List<String>) {
+            errorPatterns.forEach {
+                val matcher = it.pattern.matcher(chunk)
+                while (matcher.find()) {
+                    builder.append("${matcher.group()}$ERROR_SEPARATOR")
+                }
+            }
+            statusTags.forEach {
+                if (chunk.contains(it)) builder.append(it)
```

Comment:
`{` and `}` is needed because I am following official coding style https://kotlinlang.org/docs/reference/coding-conventions.html